pkg/cmd/helm/release/release.go

```diff
@@ -319,18 +319,26 @@ func (o *Options) OCIRegistry(repoURL, chartDir, name string) error {
 	}
 
 	if !o.NoOCILogin {
-		c = &cmdrunner.Command{
-			Dir:  chartDir,
-			Name: o.HelmBinary,
-			Env: map[string]string{
-				"HELM_EXPERIMENTAL_OCI": "1",
-			},
-			Args: []string{"registry", "login", repoURL, "--username", o.RepositoryUsername, "--password", o.RepositoryPassword},
+		if o.RepositoryUsername != "" {
+			c = &cmdrunner.Command{
+				Dir:  chartDir,
+				Name: o.HelmBinary,
+				Args: []string{"registry", "login", repoURL, "--username", o.RepositoryUsername, "--password", o.RepositoryPassword},
+			}
+
+		} else {
+			c = &cmdrunner.Command{
+				Dir:  chartDir,
+				Name: o.HelmBinary,
+				Args: []string{"registry", "login", repoURL},
+			}
+
```

Comment:
Cognitive complexity here is increased without need

o.NoOCILogin first, then else if the passworded version

Reply:
The logic has now been changed